java-generator/core/src/main/java/io/fabric8/java/generator/nodes/JDefaultValue.java

```diff
@@ -0,0 +1,60 @@
+/**
+ * Copyright (C) 2015 Red Hat, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.fabric8.java.generator.nodes;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.github.javaparser.ast.expr.Expression;
+import com.github.javaparser.ast.stmt.BlockStmt;
+import io.fabric8.java.generator.exceptions.JavaGeneratorException;
+
+public abstract class JDefaultValue<T extends AbstractJSONSchema2Pojo> {
+  protected final String key;
+  protected final String type;
+  protected final JsonNode defaultValueNode;
+
+  public JDefaultValue(String key, String type, JsonNode defaultValueNode) {
+    this.key = key;
+    this.type = type;
+    this.defaultValueNode = defaultValueNode;
+  }
+
+  public String getKey() {
+    return key;
+  }
+
+  public static JDefaultValue fromJsonSchema(String key, String type, AbstractJSONSchema2Pojo source,
```

Comment:
As discussed, we should try to leverage the tree generated already by `AbstractJSONSchema2Pojo`.

Possibly, the default value can be added as an additional field in `GeneratorResult` and expanded when needed by `JObject` during `generateJava` (I'm open to other alternatives though! 🙂  )

Reply:
In commits that are following this comment I removed the additional hierarchy that I created to represent instances of defaults, so I'll resolve this comment and move on with the latest review.